lib/puppet/network/http/api/master/v3/environment.rb

```diff
@@ -5,14 +5,15 @@ class Puppet::Network::HTTP::API::Master::V3::Environment
   def call(request, response)
     env_name = request.routing_path.split('/').last
     env = Puppet.lookup(:environments).get(env_name)
+    code_id = request.params[:code_id]
 
     if env.nil?
       raise Puppet::Network::HTTP::Error::HTTPNotFoundError.new("#{env_name} is not a known environment", Puppet::Network::HTTP::Issues::RESOURCE_NOT_FOUND)
     end
 
-    catalog = Puppet::Parser::EnvironmentCompiler.compile(env).to_resource
+    catalog = Puppet::Parser::EnvironmentCompiler.compile(env, code_id).to_resource
 
-    env_graph = {:environment => env.name, :applications => {}}
+    env_graph = {:environment => env.name, :applications => {}, :code_id => catalog.code_id}
```

Comment:
A comment above here perhaps that it gets `catalog.code_id` which may not be the same as the `code_id`. (to prevent someone messing this up by thinking `code_id` should be used...